### ISSUE TITLE
Nightly clippy fails on `array-size-threshold`

### ISSUE THREAD

**Issue by poliorcetics** [rust-lang/rust-clippy#10422]
### Summary

Using the following `clippy.toml` with nightly clippy

```toml
array-size-threshold = 10000
```

I get the following error:

```
error: error reading Clippy's configuration file `/private/tmp/toto/clippy.toml`: u128 is not supported for key `array-size-threshold`
```

### Reproducer

I tried this:

```bash
cd /tmp
cargo new toto
cd toto
echo "array-size-threshold = 10000" > clippy.toml
cargo +nightly clippy
```

I expected this to happen: No warnings, successful compilation

Instead this happened:

```
cargo +nightly clippy
    Checking toto v0.1.0 (/private/tmp/toto)
error: error reading Clippy's configuration file `/private/tmp/toto/clippy.toml`: u128 is not supported for key `array-size-threshold`

error: could not compile `toto` due to previous error
```

### Version

```text
- cargo 1.69.0-nightly (9d5b32f50 2023-02-22)
- clippy 0.1.69 (7281249a 2023-02-27)

rustc 1.69.0-nightly (7281249a1 2023-02-27)
binary: rustc
commit-hash: 7281249a19a9755e9d889ee251ec323629caadab
commit-date: 2023-02-27
host: aarch64-apple-darwin
release: 1.69.0-nightly
LLVM version: 15.0.7
```

**Comment by poliorcetics** [comment]
Also reproducible in beta

@rustbot label +regression-from-stable-to-beta
